src/OrleansRuntime/GrainDirectory/GrainDirectoryPartition.cs

```diff
@@ -40,6 +40,8 @@ internal class GrainInfo : IGrainInfo
     {
         public Dictionary<ActivationId, IActivationInfo> Instances { get; private set; }
         public int VersionTag { get; private set; }
+
+        // TODO the following field creates memory overhead for ALL grains in directory, perhaps it could be done differently
         public bool SingleInstance { get; private set; }
```

Comment:
Oops. I misunderstood - I thought this was added by this PR and did not understand why it would need to. Will revert.